src/lib/sync-profile.js

```diff
@@ -98,6 +98,19 @@ class SyncProfile {
       throw new Error('alreadySynced')
     }
 
+    const existingsProfiles = await this.masq.getProfiles()
+    const importedUsername = publicProfile.username.toUpperCase()
+    const usernameExist = existingsProfiles.find(_profile => _profile.username.toUpperCase() === importedUsername)
+
+    // HACK for tests. To test sync we create two dbs on the same device
+    if (usernameExist && process.env.NODE_ENV !== 'test') {
```

Comment:
ok, I can't think of a better solution as long as tests are run in the same browser's context